libheif/heif.cc

```diff
@@ -145,6 +145,50 @@ heif_filetype_result heif_check_filetype(const uint8_t* data, int len)
 }
 
 
+heif_filetype_result heif_check_filetype_full(const uint8_t* data, int len)
+{
```

Comment:
Proposal for a better name: `heif_is_compatible_filetype` ("is"/"has", I'm not sure).

In order to make using the function easier for the user, the input data should just be the first `len` bytes from the file.
Maybe return a `heif_error` that can tell exactly why it is not compatible or when the `len` is too small to cover the whole ftyp box.

Maybe, we can also reuse `heif_has_compatible_brand` in some way.

Reply:
> Proposal for a better name: `heif_is_compatible_filetype` ("is"/"has", I'm not sure).

I will do this. Probably "has" but will check how it reads in the caller code.

> In order to make using the function easier for the user, the input data should just be the first `len` bytes from the file. Maybe return a `heif_error` that can tell exactly why it is not compatible or when the `len` is too small to cover the whole ftyp box.

Will do. I was trying to work around the streaming read case, but in retrospect it probably breaks in other places.

> Maybe, we can also reuse `heif_list_compatible_brands` in some way.

Will check this. On first look, it seems that we don't really need the list, but could check the same underlying vector to avoid hard-coding the `supported_brands` set.

Reply:
> Maybe, we can also reuse heif_list_compatible_brands in some way.

The idea was that using `heif_list_compatible_brands`, your function could be simply calling that and checking whether a supported brand is in the returned list. Thus no more need for all the file handling or checking for incorrect parameters.

Reply:
Ah, I get it now. There would still need to be a bit more checking if you want the JPEG sanity check, but it would do most of the work. 

Would adding the JPEG check to `heif_list_compatible_brands` (i.e. returning a different error for JPEG signature) likely be acceptable?

Reply:
With the current header length check, it can never match JPEG. However I can do a check for the first four bytes matching `0xFF, 0xD8, 0xFF, 0xE0` if it would not be a valid length.

Reply:
Should now be fixed.